apps/wallet-mobile/src/NftDetails/NftDetailsImage.tsx

```diff
@@ -1,25 +1,29 @@
-import {RouteProp, useRoute} from '@react-navigation/native'
 import React from 'react'
 import {Dimensions, StyleSheet, View} from 'react-native'
 import ViewTransformer from 'react-native-easy-view-transformer'
 
 import {FadeIn} from '../components'
 import {NftPreview} from '../components/NftPreview'
 import {useMetrics} from '../metrics/metricsManager'
-import {NftRoutes} from '../navigation'
+import {NftRoutes, useParams} from '../navigation'
 import {useSelectedWallet} from '../SelectedWallet'
+import {isEmptyString} from '../utils/utils'
 import {useNft} from '../yoroi-wallets/hooks'
 
+type Params = NftRoutes['nft-details']
+
+const isParams = (params?: Params | object | undefined): params is Params => {
+  return !!params && 'id' in params && !isEmptyString(params.id)
+}
+
 export const NftDetailsImage = () => {
-  const {id} = useRoute<RouteProp<NftRoutes, 'nft-details'>>().params
+  const {id} = useParams<Params>(isParams)
```

Comment:
@jorbuedo we have `useParams` which takes a ts safe guard fn, since .params are unsafe.